tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -839,76 +839,281 @@ static PyObject * TypeError_to_NotImplemented_(PyObject* self, PyObject* args, P
+  static PythonArgParser parser(
+      {
+          "set_()",
+          "set_(Storage source)",
+          "set_(Storage source, int64_t storage_offset, IntArrayRef size, IntArrayRef stride=None)",
+          "set_(Tensor source)",
+      },
+      /*traceable=*/false);
+
+  ParsedArgs<4> parsed_args;
+  auto _r = parser.parse(args, kwargs, parsed_args);
+
+  switch (_r.idx) {
+    case 0: {
+      // aten::set_(Tensor(a!) self) -> Tensor(a!)
+      auto dispatch_set_ = [](Tensor& self) -> Tensor {
+        pybind11::gil_scoped_release no_gil;
+        return self.set_();
+      };
+      return wrap(dispatch_set_(self));
+    }
+    case 1: {
+      // aten::set_.source_Storage(Tensor(a!) self, Storage source) ->
+      // Tensor(a!)
+      at::ScalarType storage_scalar_type =
+          reinterpret_cast<THPDtype*>(
+              PyObject_GetAttrString(_r.pyobject(0), "dtype"))
+              ->scalar_type;
+      TORCH_INTERNAL_ASSERT(storage_scalar_type == self.dtype());
+      auto dispatch_set_ = [](Tensor& self, Storage source) -> Tensor {
+        pybind11::gil_scoped_release no_gil;
+        return self.set_(source);
+      };
+      return wrap(dispatch_set_(self, _r.storage(0)));
+    }
+    case 2: {
+      // aten::set_.source_Storage_storage_offset(Tensor(a!) self, Storage
+      // source, int storage_offset, int[] size, int[] stride=[]) -> Tensor(a!)
+      at::ScalarType storage_scalar_type =
+          reinterpret_cast<THPDtype*>(
+              PyObject_GetAttrString(_r.pyobject(0), "dtype"))
+              ->scalar_type;
+      TORCH_INTERNAL_ASSERT(storage_scalar_type == self.dtype());
+      auto dispatch_set_ = [](Tensor& self,
+                              Storage source,
+                              int64_t storage_offset,
+                              IntArrayRef size,
+                              IntArrayRef stride) -> Tensor {
+        pybind11::gil_scoped_release no_gil;
+        return self.set_(source, storage_offset, size, stride);
+      };
+      return wrap(dispatch_set_(
+          self, _r.storage(0), _r.toInt64(1), _r.intlist(2), _r.intlist(3)));
+    }
+    case 3: {
+      // aten::set_.source_Tensor(Tensor(a!) self, Tensor source) -> Tensor(a!)
+      auto dispatch_set_ = [](Tensor& self, const Tensor& source) -> Tensor {
+        TORCH_INTERNAL_ASSERT(source.dtype() == self.dtype());
+        pybind11::gil_scoped_release no_gil;
+        return self.set_(source);
+      };
+      return wrap(dispatch_set_(self, _r.tensor(0)));
+    }
+  }
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+
 // XXX: ops that are bound here are not exposed to the C++ api nor the JIT.
 // Any new ops added here should be accompanied with a comment why they are not
 // being registered through native_functions.yaml, and be tagged cpp / JIT
 PyMethodDef variable_methods[] = {
-  // These magic methods are all implemented on python object to wrap NotImplementedError
```

Comment:
Any substantive change here?

Reply:
No, this happened when I ran `clang-format`. I can back out the change if you want.

Reply:
This indeed caused a merge conflict while I was merging. You don't have to do anything for this PR, but for future reference, please avoid doing unnecessary reformatting like this, it greatly increases chance of merge conflicts.

Reply:
Dear reader: there was a substantive change: the addition of `set_` to the list

Reply:
Oh right, sorry I forgot about that. That goes to show why reformatting can be a bad idea.